src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -197,5 +212,38 @@ class ResourceHashtable : public ResourceObj {
   }
 };
 
+template<unsigned TABLE_SIZE, typename K, typename V>
+class FixedResourceHashtableStorage : public ResourceObj {
+  using Node = ResourceHashtableNode<K, V>;
+
+  Node* _table[TABLE_SIZE];
+protected:
+  FixedResourceHashtableStorage() {
+    memset(_table, 0, TABLE_SIZE * sizeof(Node*));
+  }
+
+  constexpr unsigned table_size() const {
+    return TABLE_SIZE;
+  }
+
+  Node** table() const {
+    return const_cast<Node**>(_table);
+  }
+};
+
+template<
+    typename K, typename V,
+    unsigned (*HASH)  (K const&)           = primitive_hash<K>,
+    bool     (*EQUALS)(K const&, K const&) = primitive_equals<K>,
+    unsigned SIZE = 256,
+    ResourceObj::allocation_type ALLOC_TYPE = ResourceObj::RESOURCE_AREA,
+    MEMFLAGS MEM_TYPE = mtInternal
+    >
+class ResourceHashtable : public ResourceHashtableBase<
+  FixedResourceHashtableStorage<SIZE, K, V>,
+    K, V, HASH, EQUALS, ALLOC_TYPE, MEM_TYPE> {
+public:
+  ResourceHashtable() : ResourceHashtableBase<FixedResourceHashtableStorage<SIZE, K, V>, K, V, HASH, EQUALS, ALLOC_TYPE, MEM_TYPE>() {}
```

Comment:
nit: can you break up this line after the V> ?

Reply:
Fixed.